src/systems/physics/Physics.cc

```diff
@@ -125,6 +127,9 @@ class ignition::gazebo::systems::PhysicsPrivate
   /// New features can't be added to this list in minor / patch releases, in
   /// order to maintain backwards compatibility with downstream physics plugins.
   public: using MinimumFeatureList = ignition::physics::FeatureList<
+          ignition::physics::SetShapeFrictionPyramidSlipCompliance,
```

Comment:
We shouldn't be adding new features to the `MinimumFeatureList` because that will break compatibility with any engine plugins that don't have that feature implemented.

Do you mind adding this in a separate list like `DetachableJointFeatureList` below, and using `entityCast`?

Reply:
Resolved here 139020c8ec0af219efd848b84b4954c2c65936e9